### PR TITLE
Add the packaging metadata to build the minio snap

### PR DESCRIPTION
**Pull request by come-maiz**
## Description
This package will let you publish the latest minio in the Ubuntu store, and from there reach many users on all the supported Ubuntu versions, and more Linux distributions in progress.

## Motivation and Context
Here at ubuntu we are working on a new packaging system. You can read more about it in https://snapcraft.io
This lets you release software to all the supported distributions without intermediaries, you can make releases on your own timeline.

## How Has This Been Tested?

Tested in Ubuntu 16.04:

    $ sudo apt install git snapcraft
    $ git clone https://github.com/elopio/minio
    $ cd minio
    $ git checkout snapcraft
    $ snapcraft
    $ sudo snap install *.snap --dangerous

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    plugs: [home, removable-media, network, network-bind]
+
+parts:
+  minio:
+    source: .
+    plugin: go
+    go-importpath: github.com/minio/minio
+    after: [go]
+  go:
+    source-tag: go1.8.3
```
Comment (harshavardhana) [inline snap/snapcraft.yaml]: Minio is not yet supported on go 1.8.x yet, can you change this to go1.7.5?

Comment (come-maiz) [reply snap/snapcraft.yaml]: Sure.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    plugs: [home, removable-media, network, network-bind]
+
+parts:
+  minio:
+    source: .
+    plugin: go
+    go-importpath: github.com/minio/minio
+    after: [go]
+  go:
+    source-tag: go1.7.5
```
Comment (harshavardhana) [inline snap/snapcraft.yaml]: We have moved to go1.8.3 now you can change this back to previous version sorry @elopio

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (donatello) [reply snap/snapcraft.yaml]: Also please add a new line at the end of the file.

**Review by donatello — APPROVED** [review]
The go version can be changed to 1.8.3 or we could do it before the next release too.

This looks good! Tested on Ubuntu 16.10.

**Comment by donatello** [comment]
Since snap is restricting access to the filesystem arbitrary paths may not be given to the minio server to run.

For simple usage:

`MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 /snap/bin/minio server $HOME/snap/minio/common/export1`

works on my laptop under my login user account. This directory is separate for minio and is allowed access.

It could also be connected up to removable media. I attached a thumbdrive that mounted to `/media/aditya/FEDORA` automatically. Then I allowed minio to access removable-media with:

`sudo snap connect minio:removable-media`

Then I ran:

`MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 /snap/bin/minio server /media/aditya/FEDORA/minio-home`

and it works!

File: snap/snapcraft.yaml (hunk 2 above)
Comment (come-maiz) [reply snap/snapcraft.yaml]: Done.

**Comment by come-maiz** [comment]
Thanks for the review @donatello !
At this point, I would suggest you to go to https://build.snapcraft.io, and try the snap in the edge channel, where it won't be visible for user searches. You can feel free to experiment there as much as you want.

Please let me know if you have any question.

**Review by harshavardhana — APPROVED** [review]

**Comment by come-maiz** [comment]
Awesome. Thanks for the merge. You have a very interesting project here, and I would love if more Ubuntu users give it a try.

Once you are done with build.snapcraft.io, you can make a call for testing in our community, like this: https://forum.snapcraft.io/t/call-for-testing-ipfs/97
And then, when you are ready to move it to stable and get more users looking at it, we can help you promoting minio. We have a very popular blog in https://insights.ubuntu.com/ I would love to see a guest post from you presenting your project.

If you need a hand with any of this, just ping me.

pura vida.

**Pull request merged**